extensions/tokenizers/rust/src/lib.rs

```diff
@@ -0,0 +1,318 @@
+    for i in ids {
+        long_ids.push(*i as jlong)
+    }
+
+    let array: jlongArray = env.new_long_array(len).unwrap();
+    env.set_long_array_region(array, 0, &long_ids).unwrap();
+    array
+}
+
+#[no_mangle]
+pub extern "system" fn Java_ai_djl_huggingface_tokenizers_jni_TokenizersLibrary_getTypeIds(
+    env: JNIEnv,
+    _: JObject,
+    handle: jlong,
+) -> jlongArray {
+    let encoding = cast_handle::<Encoding>(handle);
+    let type_ids = encoding.get_type_ids();
+    let len = type_ids.len() as jsize;
+    let mut long_ids: Vec<jlong> = Vec::new();
+    for i in type_ids {
+        long_ids.push(*i as jlong)
+    }
+
+    let array: jlongArray = env.new_long_array(len).unwrap();
+    env.set_long_array_region(array, 0, &long_ids).unwrap();
+    array
+}
+
+#[no_mangle]
+pub extern "system" fn Java_ai_djl_huggingface_tokenizers_jni_TokenizersLibrary_getWordIds(
+    env: JNIEnv,
+    _: JObject,
+    handle: jlong,
+) -> jlongArray {
+    let encoding = cast_handle::<Encoding>(handle);
+    let word_ids = encoding.get_word_ids();
+    let len = word_ids.len() as jsize;
+    let mut long_ids: Vec<jlong> = Vec::new();
+    for i in word_ids {
+        if let Some(word_id) = i {
+            long_ids.push(*word_id as jlong)
+        } else {
+            long_ids.push(-1)
+        }
+    }
+
+    let array: jlongArray = env.new_long_array(len).unwrap();
+    env.set_long_array_region(array, 0, &long_ids).unwrap();
+    array
+}
+
+#[no_mangle]
+pub extern "system" fn Java_ai_djl_huggingface_tokenizers_jni_TokenizersLibrary_getTokens(
+    env: JNIEnv,
+    _: JObject,
+    handle: jlong,
+) -> jobjectArray {
+    let encoding = cast_handle::<Encoding>(handle);
+    let tokens = encoding.get_tokens();
+    let len = tokens.len() as jsize;
+
+    let array: jobjectArray = env
+        .new_object_array(len, "java/lang/String", JObject::null())
+        .unwrap();
+    for (i, token) in tokens.iter().enumerate() {
+        let item: JString = env.new_string(&token).unwrap();
+        env.set_object_array_element(array, i as jsize, item).unwrap();
+    }
+    array
+}
+
+#[no_mangle]
+pub extern "system" fn Java_ai_djl_huggingface_tokenizers_jni_TokenizersLibrary_getAttentionMask(
+    env: JNIEnv,
+    _: JObject,
+    handle: jlong,
+) -> jlongArray {
+    let encoding = cast_handle::<Encoding>(handle);
+    let attention_masks = encoding.get_attention_mask();
+    let len = attention_masks.len() as jsize;
+    let mut long_ids: Vec<jlong> = Vec::new();
+    for i in attention_masks {
+        long_ids.push(*i as jlong)
+    }
+
+    let array: jlongArray = env.new_long_array(len).unwrap();
+    env.set_long_array_region(array, 0, &long_ids).unwrap();
+    array
+}
+
+#[no_mangle]
+pub extern "system" fn Java_ai_djl_huggingface_tokenizers_jni_TokenizersLibrary_getSpecialTokenMask(
+    env: JNIEnv,
+    _: JObject,
+    handle: jlong,
+) -> jlongArray {
+    let encoding = cast_handle::<Encoding>(handle);
```

Comment:
The pattern of transforming a Rust slice into a jlongArray kinda repeats itself here several times. It could be more DRY by making a more generic function transforming the Rust slice into a jLongArray, and accepting as input a reference to the Rust slice (e.g. special token mask).

Reply:
Agree, will address this issue in a separate PR.